electronics_model/BomBackend.py

```diff
@@ -0,0 +1,67 @@
+from typing import List, Tuple
+import os
+
+import edgir
+from edg_core import CompiledDesign, TransformUtil, BaseBackend
+
+import csv
+
+
+class BomItem:
+    def __init__(self, refdes, quantity, value, footprint, manufacturer, part_number, datasheet):
+        self.refdes = refdes
+        self.quantity = quantity
+        self.value = value
+        self.footprint = footprint
+        self.manufacturer = manufacturer
+        self.part_number = part_number
+        self.datasheet = datasheet
+
+
+class GenerateBom(BaseBackend):      # creates and populates .csv file
+    def run(self, design: CompiledDesign) -> List[Tuple[edgir.LocalPath, str]]:
+        bom_list = BomTransform(design).run()
+        name = os.path.splitext(os.path.basename(__file__))[0] + '_bom.csv'
+        with open(name, 'w', newline='') as f:
+            fieldnames = ['Ref Des', 'Quantity', 'Value', 'Footprint', 'Manufacturer', 'Part Number', 'Datasheet']
```

Comment:
Also, consider making the column headers / overall file format consistent with one of the KiCad BoM export formats, so any tools that use that format can use our output. I'd prefer using the KiCad PCB BoM generator as a simple reference, since there's so many options for the schematic BoM generator and this is what we've been using.